Jayj Eaton/wk01 - starts 7th Mar/5-fri/app.js

```diff
@@ -0,0 +1,179 @@
+
+const trainSystem = function (online, onstation, offline, offstation) {
+	let arr = [];
+	let arrEnd = [];
+	let finalArray = [];
+
+	let userOn = {};
+	let startPoint = "";
+	let userOff = {};
+	let endPoint = "";
+	lastCharEndPoint = "";
+	lastCharStartPoint = "";
+	const stopsArray = [];
+	let finalStops = [];
+	const trainLines = {
+		nLine: {
+			stations: ["Times Square n", "34th n", "28th n", "23rd n", "Union Square n", "8th n"],
+		},
+		lLine: {
+			stations: ["8th l", "6th l", "Union Square l", "3rd l", "1st l"],
+		},
+		sLine: {
+			stations: ["Grand Central s", "33rd s", "28th s", "23rd s", "Union Square s", "Astor Place s"],
+		},
+	};
+
+	const lineOn = function (online) {
+		if (online === "n") {
+			userOn = trainLines.nLine;
+			stationOn(onstation);
+		} else if (online === "l") {
+			userOn = trainLines.lLine;
+			stationOn(onstation);
+		} else if (online === "s") {
+			userOn = trainLines.sLine;
+			stationOn(onstation);
+		} else {
+			console.log("invalid train line does not exist");
+			userOn = "invalid please try a train line that exists";
+		}
+	};
+	const stationOn = function (onstation) {
+		for (let i = 0; i < userOn.stations.length; i++) {
+			let sliced = userOn.stations[i].slice(0, -2);
+
+			if (onstation !== sliced) {
+				startPoint = "not a valid station check stations for current line";
+			} else if (onstation === sliced) {
+				startPoint = userOn.stations[i];
+				lineOff(offline);
+
+				return startPoint;
+			}
+		}
+	};
+	const lineOff = function (offline) {
+		if (offline === "n") {
+			userOff = trainLines.nLine;
+			stationOff(offstation);
+		} else if (offline === "l") {
+			userOff = trainLines.lLine;
+			stationOff(offstation);
+		} else if (offline === "s") {
+			userOff = trainLines.sLine;
+			stationOff(offstation);
+		} else {
+			console.log("invalid train line does not exist");
+			userOff = "invalid please try a train line that exists";
+		}
+	};
+	const stationOff = function (offstation) {
+		for (let i = 0; i < userOff.stations.length; i++) {
+			let sliced = userOff.stations[i].slice(0, -2);
+
+			if (offstation !== sliced) {
+				endPoint = "not a valid station check stations for that line";
+			} else if (offstation === sliced) {
+				endPoint = userOff.stations[i];
+				travelLine(startPoint, endPoint);
+				return endPoint;
+			}
+		}
+	};
+	// if they change lines this function runs
+	const changedTrains = function () {
+		if (arrEnd.indexOf(endPoint) < arrEnd.indexOf(`Union Square ${lastCharEndPoint}`)) {
+			for (let i = arrEnd.indexOf(endPoint); i <= arrEnd.indexOf(`Union Square ${lastCharEndPoint}`); i++) {
+				cutStops = arrEnd[i];
+				stopsArray.push(`${cutStops}`);
+			}
+		}
+		if (arrEnd.indexOf(endPoint) > arrEnd.indexOf(`Union Square ${lastCharEndPoint}`)) {
+			console.log("here > end");
+			for (let i = arrEnd.indexOf(endPoint); i >= arrEnd.indexOf(`Union Square ${lastCharEndPoint}`); i--) {
+				cutStops = arrEnd[i];
+				stopsArray.push(`${cutStops}`);
```

Comment:
im realizing i removed the strings attatched to these litterals i probably dont need the `${}